board/safety/safety_honda.h

```diff
@@ -18,21 +25,56 @@ bool honda_bosch_hardware = false;
 bool honda_alt_brake_msg = false;
 bool honda_fwd_brake = false;
 
+static bool honda_addr_check(CAN_FIFOMailBox_TypeDef *to_push) {
+  int index = get_addr_check_index(to_push, HONDA_RX_CHECKS, HONDA_RX_CHECKS_LEN);
+  update_addr_timestamp(HONDA_RX_CHECKS, index);
+
+  if (index != -1) {
+    // checksum check
+    if (HONDA_RX_CHECKS[index].check_checksum) {
+      int checksum_byte = GET_LEN(to_push) - 1;
+      uint8_t checksum = (uint8_t)(GET_BYTE(to_push, checksum_byte) & 0xF);
+      uint8_t checksum_comp = 0U;
+      unsigned int addr = GET_ADDR(to_push);
+      while (addr > 0U) {
+        checksum_comp += (addr & 0xFU); addr >>= 4;
+      }
+      for (int j = 0; j <= checksum_byte; j++) {
+        uint8_t byte = GET_BYTE(to_push, j);
+        checksum_comp += (byte & 0xFU) + (byte >> 4U);
+      }
+      checksum_comp -= checksum;  // remove checksum in message
+      checksum_comp = (8U - checksum_comp) & 0xFU;
+      HONDA_RX_CHECKS[index].valid_checksum = checksum_comp == checksum;
+    }
+
+    // get counter
+    if (HONDA_RX_CHECKS[index].max_counter > 0U) {
+      int counter_byte = GET_LEN(to_push) - 1;
+      uint8_t counter = ((uint8_t)(GET_BYTE(to_push, counter_byte)) >> 4U) & 0x3U;
+      update_counter(HONDA_RX_CHECKS, index, counter);
+    }
+  }
+  return is_msg_valid(HONDA_RX_CHECKS, index);
+}
+
 static void honda_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
 
   int addr = GET_ADDR(to_push);
   int len = GET_LEN(to_push);
   int bus = GET_BUS(to_push);
 
+  bool valid = honda_addr_check(to_push);
+
   // sample speed
-  if (addr == 0x158) {
+  if ((addr == 0x158) && valid) {
```

Comment:
Why check for valid everywhere. Can we just skip the whole function if the message is not valid?